src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -35,14 +39,25 @@ protected internal override void Setup(FeatureConfigurationContext context)
             var failureInfoStorage = new FailureInfoStorage(context.Settings.Get<int>(FailureInfoStorageCacheSizeKey));
             var localAddress = context.Settings.LocalAddress();
 
-            context.Pipeline.Register("MoveFaultsToErrorQueue", b => new MoveFaultsToErrorQueueBehavior(
-                b.Build<CriticalError>(),
-                localAddress,
-                transportTransactionMode,
-                failureInfoStorage), "Moves failing messages to the configured error queue");  //context.Pipeline.Register(new MoveFaultsToErrorQueueBehavior.Registration(context.Settings.LocalAddress(), transportTransactionMode, failureInfoStorage));
-
-            context.Pipeline.Register("AddExceptionHeaders", new AddExceptionHeadersBehavior(), "Adds the exception headers to the message");
-            context.Pipeline.Register(new FaultToDispatchConnector(errorQueue), "Connector to dispatch faulted messages");
+            context.Pipeline.Register("MoveFaultsToErrorQueue", b =>
+            {
+                var hostInfo = b.Build<HostInformation>();
+                var staticFaultMetadata = new Dictionary<string, string>
+                {
+                    {FaultsHeaderKeys.FailedQ, localAddress},
+                    {Headers.ProcessingMachine, RuntimeEnvironment.MachineName },
+                    {Headers.ProcessingEndpoint, context.Settings.EndpointName()},
+                    {Headers.HostId, hostInfo.HostId.ToString("N")},
+                    {Headers.HostDisplayName, hostInfo.DisplayName}
```

Comment:
while I like that approach, I thought we also wanted these headers to be present on satellite error messages?